extensions/default/src/DicomTagBrowser/DicomTagTable.tsx

```diff
@@ -1,97 +1,203 @@
-            <span className="flex flex-row items-center focus:outline-none">
-              Tag
-            </span>
-          </label>
-        </div>
-        <div className="px-3 w-4/24">
-          <label className="flex flex-col flex-1 text-white text-lg pl-1 select-none">
-            <span className="flex flex-row items-center focus:outline-none">
-              VR
-            </span>
-          </label>
-        </div>
-        <div className="px-3 w-6/24">
-          <label className="flex flex-col flex-1 text-white text-lg pl-1 select-none">
-            <span className="flex flex-row items-center focus:outline-none">
-              Keyword
-            </span>
-          </label>
-        </div>
-        <div className="px-3 w-5/24">
-          <label className="flex flex-col flex-1 text-white text-lg pl-1 select-none">
-            <span className="flex flex-row items-center focus:outline-none">
-              Value
-            </span>
-          </label>
-        </div>
+    <div
+      className={classNames(
+        'flex flex-row w-full bg-secondary-dark ohif-scrollbar overflow-y-scroll'
+      )}
+      style={rowVerticalPaddingStyle}
+    >
+      <div className="px-3 w-4/24">
+        <label
+          ref={tagRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            Tag
+          </span>
+        </label>
+      </div>
+      <div className="px-3 w-2/24">
+        <label
+          ref={vrRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            VR
+          </span>
+        </label>
+      </div>
+      <div className="px-3 w-6/24">
+        <label
+          ref={keywordRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            Keyword
+          </span>
+        </label>
+      </div>
+      <div className="px-3 w-5/24 grow">
+        <label
+          ref={valueRef}
+          className="flex flex-col flex-1 text-white text-lg pl-1 select-none"
+        >
+          <span className="flex flex-row items-center focus:outline-none">
+            Value
+          </span>
+        </label>
       </div>
     </div>
   );
 }
 
 function DicomTagTable({ rows }) {
+  const listRef = useRef();
+  const canvasRef = useRef();
+
+  /**
+   * When new rows are set, scroll to the top and reset the virtualization.
+   */
+  useEffect(() => {
+    if (!listRef?.current) {
+      return;
+    }
+
+    listRef.current.scrollTo(0);
+    listRef.current.resetAfterIndex(0);
+  }, [rows]);
+
+  /**
+   * When the browser window resizes, update the row virtualization (i.e. row heights)
+   */
+  React.useEffect(() => {
+    function handleResize() {
```

Comment:
handleResize could (maybe should because of useEffect dependency thingy in react) be outside

Reply:
I think once I add the `[]` dependency, this effect should be fine.